src/main/java/sirius/db/jdbc/BaseSQLQuery.java

```diff
@@ -95,8 +95,9 @@ public void iterateAll(Consumer<Row> consumer, @Nullable Limit limit) throws SQL
     protected void processResultSet(Predicate<Row> handler,
                                     Limit effectiveLimit,
                                     ResultSet resultSet,
-                                    TaskContext taskContext) throws SQLException {
-        while (resultSet.next() && taskContext.isActive()) {
+                                    TaskContext taskContext,
+                                    boolean longRunning) throws SQLException {
+        while (resultSet.next() && (taskContext.isActive() && longRunning)) {
```

Comment:
shoudln't we factor this out here entirely (we unvoke a handler returning a boolean anyway) - this way, we could only enforce this in interate, iterateAll and blockwise* without having the extra complexity in here. We might even handle this on a database independent level. Also "longRunning" isn't an ideal name, as otherwise "long running queries" are such, where the execution itself might be long running (even before results are seen) and thus should not belogged. This is however no indication on the result set size.